modules/tinymce/src/core/main/ts/mode/Disabled.ts

```diff
@@ -0,0 +1,144 @@
+    ModeUtils.enableEditor(editor);
+  }
+};
+
+const registerDisabledContentFilters = (editor: Editor): void => {
+  if (editor.serializer) {
+    registerFilters(editor);
+  } else {
+    editor.on('PreInit', () => {
+      registerFilters(editor);
+    });
+  }
+};
+
+const registerFilters = (editor: Editor): void => {
+  editor.parser.addAttributeFilter('contenteditable', (nodes) => {
+    if (isDisabled(editor)) {
+      Arr.each(nodes, (node) => {
+        node.attr(internalContentEditableAttr, node.attr('contenteditable'));
+        node.attr('contenteditable', 'false');
+      });
+    }
+  });
+
+  editor.serializer.addAttributeFilter(internalContentEditableAttr, (nodes) => {
+    if (isDisabled(editor)) {
+      Arr.each(nodes, (node) => {
+        node.attr('contenteditable', node.attr(internalContentEditableAttr));
+      });
+    }
+  });
+
+  editor.serializer.addTempAttr(internalContentEditableAttr);
+};
+
+const isClickEvent = (e: Event): e is MouseEvent => e.type === 'click';
+
+const allowedEvents: ReadonlyArray<string> = [ 'copy' ];
+
+const isDisabledAllowedEvent = (e: Event) => Arr.contains(allowedEvents, e.type);
+
+const getAnchorHrefOpt = (editor: Editor, elm: SugarElement<Node>): Optional<string> => {
+  const isRoot = (elm: SugarElement<Node>) => Compare.eq(elm, SugarElement.fromDom(editor.getBody()));
+  return SelectorFind.closest<HTMLAnchorElement>(elm, 'a', isRoot).bind((a) => Attribute.getOpt(a, 'href'));
+};
+
+const processDisabledEvents = (editor: Editor, e: Event): void => {
+  /*
+    If an event is a click event on or within an anchor, and the CMD/CTRL key is
+    not held, then we want to prevent default behaviour and either:
+      a) scroll to the relevant bookmark
+      b) open the link using default browser behaviour
+  */
+  if (isClickEvent(e) && !VK.metaKeyPressed(e)) {
+    const elm = SugarElement.fromDom(e.target as Node);
+    getAnchorHrefOpt(editor, elm).each((href) => {
+      e.preventDefault();
+      if (/^#/.test(href)) {
+        const targetEl = editor.dom.select(`${href},[name="${Strings.removeLeading(href, '#')}"]`);
+        if (targetEl.length) {
+          editor.selection.scrollIntoView(targetEl[0], true);
+        }
+      } else {
+        window.open(href, '_blank', 'rel=noopener noreferrer,menubar=yes,toolbar=yes,location=yes,status=yes,resizable=yes,scrollbars=yes');
+      }
+    });
+  } else if (isDisabledAllowedEvent(e)) {
+    editor.dispatch(e.type, e);
+  }
+};
+
+const registerDisabledModeEventHandlers = (editor: Editor): void => {
+  editor.on('ShowCaret', (e) => {
+    if (isDisabled(editor)) {
+      e.preventDefault();
+    }
+  });
+
+  editor.on('ObjectSelected', (e) => {
+    if (isDisabled(editor)) {
+      e.preventDefault();
+    }
+  });
+
+  // Preprend to the handlers as this should be the first to fire
+  editor.on('DisabledStateChange', (e) => {
+    if (!e.isDefaultPrevented()) {
+      toggleDisabled(editor, e.state);
+    }
+  }, true);
```

Comment:
This seems a bit fragile. Should it be moved to `Options`, or should it be checked by `e.state` instead of `editor.disabled` in the rest of the places?

Reply:
Previously, the `editor.readonly` state is updated before the event is fired, so we can use `editor.readonly` in the event listener for the rest of the places. 
https://github.com/tinymce/tinymce/blob/908f8c9dcb409b0ff44fcc6cb5cda52cf52b86f0/modules/tinymce/src/core/main/ts/mode/Mode.ts#L24
https://github.com/tinymce/tinymce/blob/908f8c9dcb409b0ff44fcc6cb5cda52cf52b86f0/modules/tinymce/src/core/main/ts/mode/Readonly.ts#L57